skyvern/webeye/browser_manager.py

```diff
@@ -66,18 +67,51 @@ def get_for_task(self, task_id: str, workflow_run_id: str | None = None) -> Brow
 
         return None
 
-    async def get_or_create_for_task(self, task: Task) -> BrowserState:
+    async def get_or_create_for_task(
+        self,
+        task: Task,
+        browser_session_id: str | None = None,
+    ) -> BrowserState:
         browser_state = self.get_for_task(task_id=task.task_id, workflow_run_id=task.workflow_run_id)
         if browser_state is not None:
             return browser_state
 
-        LOG.info("Creating browser state for task", task_id=task.task_id)
-        browser_state = await self._create_browser_state(
-            proxy_location=task.proxy_location,
-            url=task.url,
-            task_id=task.task_id,
-            organization_id=task.organization_id,
-        )
+        if browser_session_id:
+            LOG.info(
+                "Getting browser state for task from persistent sessions manager",
+                browser_session_id=browser_session_id,
+            )
+            browser_state = app.PERSISTENT_SESSIONS_MANAGER.get_browser_state(browser_session_id)
+            if browser_state is None:
+                LOG.warning(
+                    "Browser state not found in persistent sessions manager",
+                    browser_session_id=browser_session_id,
+                )
+            else:
+                if task.organization_id:
+                    await app.PERSISTENT_SESSIONS_MANAGER.occupy_browser_session(
+                        browser_session_id,
+                        organization_id=task.organization_id,
+                        runnable_type="task",
+                        runnable_id=task.task_id,
+                    )
+                else:
+                    LOG.warning("Organization ID is not set for task", task_id=task.task_id)
+                page = await browser_state.get_working_page()
+                if page:
+                    await browser_state.navigate_to_url(page=page, url=task.url)
+                else:
+                    LOG.warning("Browser state has no page", workflow_run_id=task.workflow_run_id)
+
+        if browser_state is None:
```

Comment:
if there's a browser_session_id but browser_state is still None here, i think we can raise error and not continue for now.

Reason: we can do the local browser fallback later. If we expect to connect to a browser session to run the agent, spinning up a local browser will likely hide bugs. Secondly, i'm not sure if it's true but it's more likely to introduce configuration errors when you're expecting the config is for a persistent browser session but actually using a local browser.

Reply:
right, makes sense 👍